google/cloud/bigquery/client.py

```diff
@@ -890,14 +961,20 @@ def update_routine(self, routine, fields, retry=DEFAULT_RETRY, timeout=None):
         # TODO: remove when routines update supports partial requests.
         partial["routineReference"] = routine.reference.to_api_repr()
 
-        api_response = self._call_api(
-            retry,
-            method="PUT",
-            path=routine.path,
-            data=partial,
-            headers=headers,
-            timeout=timeout,
-        )
+        path = routine.path
+        span_attributes = {"path": path, "fields": fields}
```

Comment:
fields is a list of strings. Serialize to JSON string?

Reply:
Will do